### DISCUSSION TITLE
Observable + Leaflet App

### DISCUSSION THREAD

**Discussion by kbvernon** [quarto-dev/quarto-cli#4100] [Show and Tell]
I made a simple (simple 😂) Observable + Leaflet app for exploring watersheds in the US Southwest (mostly New Mexico and Arizona). The [APP IS HERE](https://kbvernon.github.io/hndsr-watersheds/) and the [CODE IS HERE](https://github.com/kbvernon/hndsr-watersheds). 

The main thing I struggled with is getting cells to communicate in both directions, so to speak, without circularity. That took me a minute to figure out as reactivity in Observable is utterly mystifying, especially the concept of mutation and mutable cells (which are available but frowned upon), but maybe that's because JavaScript for me might as well be [Tamarian](https://memory-alpha.fandom.com/wiki/Darmok_(episode)). 

Anyway, I have a lot to say about the experience, but I'll leave it at that.

As always, thanks to the Posit team for making Quarto! It's such a great tool, I find myself constantly fighting the urge to believe that it's really me that's being clever. 🥸

**Comment by cscheid** [comment]
Hey, that's a very cool demo, thanks for sharing!

**Comment by edenian-prince** [comment]
this is awesome. im struggling with observable rn and am going to use this as a reference

**Reply by kbvernon** [reply]
thanks for saying that! hopefully it helps!
